### ISSUE TITLE
[PRE REVIEW]: mixR: An R package for Finite Mixture Modeling for Both Raw and Binned Data

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#4028]
**Submitting author:** @GaryBAYLOR (<a href="http://orcid.org/0000-0003-0519-9605">Youjiao Yu</a>)
**Repository:** <a href="https://github.com/GaryBAYLOR/mixR.git" target ="_blank">https://github.com/GaryBAYLOR/mixR.git</a>
**Version:** 0.2.0
**Editor:** @fboehm
**Reviewers:** @welch16, @soodoku
**Managing EiC:** Arfon Smith

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/f330cc88d1d8dc43fb6ba8f426beff21/status.svg)](https://joss.theoj.org/papers/f330cc88d1d8dc43fb6ba8f426beff21)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/f330cc88d1d8dc43fb6ba8f426beff21"><img src="https://joss.theoj.org/papers/f330cc88d1d8dc43fb6ba8f426beff21/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/f330cc88d1d8dc43fb6ba8f426beff21/status.svg)](https://joss.theoj.org/papers/f330cc88d1d8dc43fb6ba8f426beff21)
```

**Author instructions**

Thanks for submitting your paper to JOSS @GaryBAYLOR. **Currently, there isn't an JOSS editor assigned** to your paper.

@GaryBAYLOR if you have any suggestions for potential reviewers then please mention them here in this thread (without tagging them with an @). In addition, [this list of people](https://bit.ly/joss-reviewers) have already agreed to review for JOSS and may be suitable for this submission (please start at the bottom of the list).

**Editor instructions**

The JOSS submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
Wordcount for `paper.md` is 1138

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.06 s (1058.7 files/s, 109047.8 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
R                               51            412            981           2064
C++                              2            184            271           1602
TeX                              2             30              0            233
Markdown                         4             60              0            163
Rmd                              1             51            126             69
YAML                             1              4              7             26
-------------------------------------------------------------------------------
SUM:                            61            741           1385           4157
-------------------------------------------------------------------------------


Statistical information for the repository '944586b76923fa0a5944588a' was
gathered on 2021/12/29.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Xiaozhen Han                     1            40              0            1.91
Youjiao (Gary) Yu                1          2035             18           98.09

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Xiaozhen Han                 22           55.0          0.0                9.09
Youjiao (Gary) Yu          2035          100.0          0.0               14.59
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.04028/joss.04028/10.21105.joss.04028.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.04028/joss.04028/10.21105.joss.04028.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- None

MISSING DOIs

- 10.1111/j.2517-6161.1996.tb02104.x may be a valid DOI for title: Using bootstrap likelihood ratios in finite mixture models
- 10.2307/2347790 may be a valid DOI for title: On bootstrapping the likelihood ratio test statistic for the number of components in a normal mixture
- 10.1080/03610926.2018.1494838 may be a valid DOI for title: Bootstrap likelihood ratio test for Weibull mixture models fitted to grouped data
- 10.32614/rj-2016-021 may be a valid DOI for title: mclust 5: clustering, classification and density estimation using Gaussian finite mixture models
- 10.18637/jss.v011.i08 may be a valid DOI for title: Flexmix: A general framework for finite mixture models and latent glass regression in R
- 10.1007/s10661-020-8223-4 may be a valid DOI for title: Benchmarking inference methods for water quality monitoring and status classification
- 10.1016/j.watres.2019.115369 may be a valid DOI for title: Can routine monitoring of E. coli fully account for peak event concentrations at drinking water intakes in agricultural and urban rivers?
- 10.1080/10549811.2020.1841009 may be a valid DOI for title: Does the Inclusion of Truncation point in a Finite Mixture Model Improve Diameter Distribution Estimation of Degraded Stand?
- 10.1101/2020.10.21.348573 may be a valid DOI for title: Clustering and erratic movement patterns of syringe-injected versus mosquito-inoculated malaria sporozoites underlie decreased infectivity
- 10.1016/j.jvolgeores.2021.107257 may be a valid DOI for title: Measuring the size of non-spherical particles and the implications for grain size analysis in volcanology
- 10.1016/j.physa.2021.126764 may be a valid DOI for title: Modeling conditional dependencies for bus travel time estimation
- 10.1093/forsci/fxab051 may be a valid DOI for title: Characterizing Stand and Biomass Tables from Diameter Distribution Models: A Case Study for Mixed-Hardwood Forests in Eastern Tennessee, USA

INVALID DOIs

- None
```

**Comment by arfon** [comment]
@whedon invite @fboehm as editor

:wave: @fboehm – would you be willing to edit this submission for JOSS?

**Comment by whedon** [comment]
@fboehm has been invited to edit this submission.

**Comment by arfon** [comment]
@GaryBAYLOR – many thanks for your submission to JOSS. While we look for an editor, if you want to fix the DOIs (noting that whedon's suggestions are not always right), you can, then use the following commands (one at a time, as the first line of a new comment) to regenerate the PDF and check the references.

@whedon generate pdf
@whedon check references

**Comment by fboehm** [comment]
@whedon assign @fboehm as editor

**Comment by whedon** [comment]
OK, the editor is @fboehm

**Comment by fboehm** [comment]
@arfon - yes, I'm happy to edit this submission. I'll start looking for reviewers. Thanks!

**Comment by GaryBAYLOR** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.04028/joss.04028/10.21105.joss.04028.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.04028/joss.04028/10.21105.joss.04028.pdf) :page_facing_up: :point_left:

**Comment by GaryBAYLOR** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1111/j.2517-6161.1977.tb01600.x is OK
- 10.1111/j.2517-6161.1996.tb02104.x is OK
- 10.2307/2347790 is OK
- 10.1080/03610926.2018.1494838 is OK
- 10.18637/jss.v032.i06 is OK
- 10.32614/rj-2016-021 is OK
- 10.18637/jss.v011.i08 is OK
- 10.18637/jss.v040.i08 is OK
- 10.1002/wics.147 is OK
- 10.1007/s10661-020-8223-4 is OK
- 10.1016/j.watres.2019.115369 is OK
- 10.1080/10549811.2020.1841009 is OK
- 10.1101/2020.10.21.348573 is OK
- 10.1016/j.jvolgeores.2021.107257 is OK
- 10.1016/j.physa.2021.126764 is OK
- 10.1093/forsci/fxab051 is OK

MISSING DOIs

- None

INVALID DOIs

- http://dx.doi.org/10.1142/12366 is INVALID because of 'https://doi.org/' prefix
```

**Comment by GaryBAYLOR** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1111/j.2517-6161.1977.tb01600.x is OK
- 10.1111/j.2517-6161.1996.tb02104.x is OK
- 10.2307/2347790 is OK
- 10.1080/03610926.2018.1494838 is OK
- 10.18637/jss.v032.i06 is OK
- 10.32614/rj-2016-021 is OK
- 10.18637/jss.v011.i08 is OK
- 10.18637/jss.v040.i08 is OK
- 10.1002/wics.147 is OK
- 10.1007/s10661-020-8223-4 is OK
- 10.1016/j.watres.2019.115369 is OK
- 10.1080/10549811.2020.1841009 is OK
- 10.1101/2020.10.21.348573 is OK
- 10.1016/j.jvolgeores.2021.107257 is OK
- 10.1016/j.physa.2021.126764 is OK
- 10.1093/forsci/fxab051 is OK
- 10.1142/12366 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by GaryBAYLOR** [comment]
Thanks! @arfon @fboehm I just added DOIs to all references.

**Comment by fboehm** [comment]
@whedon list reviewers

**Comment by whedon** [comment]
Here's the current list of reviewers: https://bit.ly/joss-reviewers

**Comment by fboehm** [comment]
@paragkulkarni11, @ChristopherLucas, @Medewitt, @soodoku - would any of you be willing to review this submission for JOSS? Thanks!

**Comment by fboehm** [comment]
@whedon assign @welch16 as reviewer

**Comment by whedon** [comment]
OK, @welch16 is now a reviewer

**Comment by soodoku** [comment]
hey @GaryBAYLOR: can you say more about the diff. w/ mixtools? what does your package do that mixtools doesn't?

**Comment by GaryBAYLOR** [comment]
Hi @soodoku, Thanks for the question. The main features in mixR but not in mixtools package are
(a) `mixR` can fit mixtures models for both raw data and binned data, while `mixtools` is only for raw data (which might not be a very precise summary as `mixtools` provides functions to analyze discrete data such as Poisson data, multinomial data, as well as allowing censored data. But since mixR focuses on mixtures of continuous distributions, e.g., Gaussian, Weibull, Log-normal, Gamma, if we restrict comparison to the mixtures models of continuous distributions then `mixtools` uses raw data and for some mixtures allowing right censored data, as model inputs) 

(b) `mixR` uses a single interface function (_mixfit_) for model fitting for both types of data and all distributions, while `mixtools` provides separate functions for modeling different distributions, such as _normalmixEM_, _normalmixEM2comp_, _gammamixEM_, _weibullRMM_SEM_, etc.

(c) Other features such as 
      - `mixR` provides mixtures of log-normal 
      - `mixR` improves the visualizations using `ggplot2` with flexibility to customize the output to easily produce publication-quality plots
      - `mixR` provides functions to convert parameters between shape-scale format to mean-variance format for Weibull and Gamma.

In summary, (a) is the primary contribution of `mixR` compared to `mixtools`, (b) is the implementation that simplifies how to use the package, (c) are handy features that users often need.

NOTES:
The raw data refers to numeric data x_1, x_2, ..., x_n. The binned data (sometimes called grouped data), is defined as (a_1, b_1, k_1), (a_2, b_2, k_2), ... in which a_i and b_i are cutoff points that form an interval, and k_i is the count of data points that fall in (a_i, b_i). This is the definition in univariate case but the definition can be extended to multivariate data.

Please let me know any other questions for you to better understand and evaluate `mixR`.

Thank you!
Youjiao

**Comment by soodoku** [comment]
thanks for the summary @GaryBAYLOR.  i also read the paper.  @fboehm happy to review.

**Comment by fboehm** [comment]
@whedon add @soodoku as reviewer

**Comment by whedon** [comment]
OK, @soodoku is now a reviewer

**Comment by fboehm** [comment]
@whedon begin review

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by fboehm** [comment]
@whedon start review

**Comment by whedon** [comment]
OK, I've started the review over in https://github.com/openjournals/joss-reviews/issues/4031.
